frontend/src/views/Lobby.tsx

```diff
@@ -265,13 +296,33 @@ function LobbyPage() {
         <DifficultyButton
           onClick={() => updateDifficultySetting(key)}
           active={difficulty === Difficulty[key as keyof typeof Difficulty]}
-          enabled={currentUser?.nickname === host?.nickname}
+          enabled={isHost(currentUser)}
           title={currentUser?.nickname !== host?.nickname
             ? 'Only the host can change these settings' : undefined}
         >
           {key}
         </DifficultyButton>
       ))}
+
+      <MediumText>Duration</MediumText>
+      <Text>
+        {isHost(currentUser) ? 'Choose a game duration between 1-60 minutes:'
+          : 'The game will last for the following minutes:'}
+      </Text>
+      <NumberInput
+        min={1}
+        max={60}
+        value={duration}
+        disabled={!isHost(currentUser)}
+        onChange={(e) => {
+          const newDuration = Number(e.target.value);
+          if (newDuration >= 0 && newDuration <= 60) {
+            setDuration(Number(e.target.value));
+          }
+        }}
+      />
+      {isHost(currentUser) ? <SmallButton onClick={updateRoomDuration}>Save</SmallButton> : null}
```

Comment:
Just wanted to bring up that we could potentially not use the "Save" button, and just auto-save on every change, **or** just take the current value of the field when "Start Game" is pressed.

Another idea would be to auto-save the value whenever the user clicks away from the input again. Thoughts on these alternative approaches? Obviously it's no big deal for the MVP, but might be nice in the future to have a cleaner frontend system from which to build & design more.

Reply:
Good idea on that - I ended up going with the autosave when you click anywhere away from it (onBlur) for a balance between too often and too little saving